pkg/api/handlers/compat/containers_top.go

```diff
@@ -48,12 +48,8 @@ func TopContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// We are committed now - all errors logged but not reported to client, ship has sailed
-	w.WriteHeader(http.StatusOK)
+	statusWritten := false
 	w.Header().Set("Content-Type", "application/json")
-	if f, ok := w.(http.Flusher); ok {
```

Comment:
I'm not real sure why this was in here, nor am I sure as to why it was removed.  I think it wasn't necessary,  do you know what it was here for?

Reply:
The problem is that once 200 was sent, we cannot handle errors anymore.  So before, all (most?) requests returned 200 even in case of an error.